### PR TITLE
Fix: Solution for the issues with piping returning null

### LINKED ISSUE #61: [Bug]: Unexpected NULL return in `data-name` attribute when using `%>%` operator with 'validate()' function
### Guidelines

- [X] I agree to follow this project's Contributing Guidelines.

### Project Version

0.2.0

### Platform and OS Version

macOs 13.3.1

### What happened?

When `name` argument is not provided, the function should return the name of `data` object as `data-name` attribute. This functionality works as expected when using either `validate(data)` or `data |> validate()`. However, when attempting to use `%>%` operator, the function returns NULL instead of the expected data name.

### Steps to reproduce

```{r}
library(data.validator)
library(dplyr)

data <- data.frame(V1 = c(0,1,2), V2 = c(3,4,5))
data_val <- validate(data)
```

First two examples returning name of the data object:

```{r}
# This works as expected
> attr(validate(data_val), "data-name")
[1] "data_val"
>
# This also works as expected
> data_val |>
+     validate() |>
+     attr("data-name")
[1] "data_val"
```

`NULL` return when using `%>%`:

```
> data_val %>%
+     validate() %>%
+     attr("data-name")
[1] "NULL"
```
### Expected behavior

When using the `%>%` operator with the `validate()` function, it should return the correct `data-name` attribute, similar to the other valid usage examples mentioned above.

### Attachments

```
> sessionInfo()
R version 4.2.1 (2022-06-23)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] assertr_3.0.0        dplyr_1.0.8          data.validator_0.2.0
```

### PR DESCRIPTION
**Pull request by wrprates**
### Link to the issue
Closes #61 

Main changes:

- Updated functionality: The `get_first_name` function was revised to detect the data frame name in `magrittr` pipes (`%>%`). This now allows the `validate` function to correctly assign the "data-name" attribute when using `%>%`.
- New internal functions: added `get_ast`, `find_first_noncall`, and `is_complex_command` functions to read and understand the abstract syntax tree (AST) of a function call (check [this](https://community.rstudio.com/t/using-deparse-substitute-expression-with-pipe/137595/2) discussion), identify `magrittr` pipes, and provide the name of the starting object in a pipe sequence.
- Test coverage: added unit tests for the new functions and revised `get_first_name` function to ensure correct operation across various use cases.

### CODE REVIEW

**Review by jakubnowicki — COMMENTED** [review]
I really like the solution :+1: 
Please bump package version and update NEWS.md file.

File: tests/testthat/testthat-problems.rds
Comment (jakubnowicki) [inline tests/testthat/testthat-problems.rds:1]: This file seems to be redundant. Please check and if yes - remove it.

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # data.validator (development version)
 
 - Swapping of `error` and `warning` arguments in `save_summary()` fixed
+- Enhanced `validate()` function to correctly return `data-name` attribute when used in pipe chains with `%>%` or `|>` operator.
```
Comment (jakubnowicki) [inline NEWS.md:4]:
```suggestion
- Fixed`validate()` function to correctly return `data-name` attribute when used in pipe chains with `%>%` or `|>` operator.
```

**Review by jakubnowicki — APPROVED** [review]
Good job, please resolve my comments and you can merge.

File: tests/testthat/testthat-problems.rds
Comment (wrprates) [reply tests/testthat/testthat-problems.rds:1]: Removed.

**Pull request merged**